src/Transports.AspNetCore.SystemTextJson/InternalGraphQLRequest.cs

```diff
@@ -19,13 +18,13 @@ internal sealed class InternalGraphQLRequest
         /// <see cref="GraphQL.SystemTextJson.ObjectDictionaryConverter"/>.
         /// </remarks>
         [JsonPropertyName(GraphQLRequest.VARIABLES_KEY)]
-        public Dictionary<string, object> Variables { get; set; }
+        public Inputs Variables { get; set; }
 
         /// <remarks>
         /// Population of this property during deserialization from JSON requires
         /// <see cref="GraphQL.SystemTextJson.ObjectDictionaryConverter"/>.
         /// </remarks>
         [JsonPropertyName(GraphQLRequest.EXTENSIONS_KEY)]
-        public Dictionary<string, object> Extensions { get; set; }
+        public Inputs Extensions { get; set; }
```

Comment:
😄  I expect you do exactly that change. But what about nested dictionaries (not inputs) inside that top level property? Will they be properly parsed?

Reply:
Yes, but I should add a test to be sure.

Reply:
Done